tests/role_parsing_test.go

```diff
@@ -0,0 +1,73 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+package test
+
+import (
+	"encoding/json"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/ecs/types"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+)
+
+// TestRoleParsingWithPath tests that the module correctly parses role names from ARNs with paths
+func (s *ECSFargateSuite) TestRoleParsingWithPath() {
+	log.Println("TestRoleParsingWithPath: Running test...")
+
+	var containers []types.ContainerDefinition
+	task := terraform.OutputMap(s.T(), s.terraformOptions, "role-parsing-with-path")
+
+	s.Equal(s.testPrefix+"-role-parsing-with-path", task["family"], "Unexpected task family name")
+
+	err := json.Unmarshal([]byte(task["container_definitions"]), &containers)
+	s.NoError(err, "Failed to parse container definitions")
+
+	s.NotEmpty(task["arn"], "Task definition ARN should not be empty")
+	s.NotEmpty(task["revision"], "Task definition revision should not be empty")
+
+	taskRoleArn := task["task_role_arn"]
+	s.NotEmpty(taskRoleArn, "Task role ARN should not be empty")
+	s.Contains(taskRoleArn, "/terraform-test/", "Task role ARN should contain the path '/test-path/'")
```

Comment:
nit: expected role path
```suggestion
	s.Contains(taskRoleArn, "/terraform-test/", "Task role ARN should contain the path '/terraform-test/'")
```